src/sg/aca_security_group.cpp

```diff
@@ -0,0 +1,353 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "aca_security_group.h"
+
+using namespace std;
+
+
+namespace aca_security_group {
+
+void Aca_Security_Group_Rule::set_id(string id) {
+	this->id = id;
+}
+
+string Aca_Security_Group_Rule::get_id(void) {
+	return this->id;
+}
+
+void Aca_Security_Group_Rule::set_name(string name) {
+	this->name = name;
+}
+
+
+string Aca_Security_Group_Rule::get_name(void) {
+	return this->name;
+}
+
+void Aca_Security_Group_Rule::set_cookie(uint64_t cookie) {
+	this->cookie = cookie;
+}
+
+uint64_t Aca_Security_Group_Rule::get_cookie(void) {
+	return this->cookie;
+}
+
+void Aca_Security_Group_Rule::set_direction(Direction direction) {
+	this->direction = direction;
+}
+
+Direction Aca_Security_Group_Rule::get_direction(void) {
+	return this->direction;
+}
+
+void Aca_Security_Group_Rule::set_ethertype(Ethertype ethertype) {
+	this->ethertype = ethertype;
+
+}
+	
+Ethertype Aca_Security_Group_Rule::get_ethertype(void) {
+	return this->ethertype;
+}
+
+void Aca_Security_Group_Rule::set_protocol(Protocol protocol) {
+	this->protocol = protocol;
+
+}
+Protocol Aca_Security_Group_Rule::get_protocol(void) {
+	return this->protocol;
+}
+
+void Aca_Security_Group_Rule::set_port_range_min(uint32_t port_range_min) {
+	this->port_range_min = port_range_min;
+}
+
+uint32_t Aca_Security_Group_Rule::get_port_range_min(void) {
+	return this->port_range_min;
+}
+
+void Aca_Security_Group_Rule::set_port_range_max(uint32_t port_range_max) {
+	this->port_range_max = port_range_max;
+}
+
+uint32_t Aca_Security_Group_Rule::get_port_range_max(void) {
+	return this->port_range_max;
+}
+
+void Aca_Security_Group_Rule::set_remote_ip_prefix(string remote_ip_prefix) {
+	this->remote_ip_prefix = remote_ip_prefix;
+}
+
+string Aca_Security_Group_Rule::get_remote_ip_prefix(void) {
+	return this->remote_ip_prefix;
+}
+
+void Aca_Security_Group_Rule::set_remote_group_id(string remote_group_id) {
+	this->remote_group_id = remote_group_id;
+}
+
+string Aca_Security_Group_Rule::get_remote_group_id(void) {
+	return this->remote_group_id;
+}
+
+void Aca_Security_Group_Rule::set_operation_type(OperationType operation_type) {
+	this->operation_type = operation_type;
+}
+
+OperationType Aca_Security_Group_Rule::get_operation_type(void) {
+	return this->operation_type;
+}
+
+void Aca_Security_Group_Rule::set_remote_group(Aca_Security_Group * remote_group) {
+	this->remote_group = remote_group;
+}
+
+Aca_Security_Group * Aca_Security_Group_Rule::get_remote_group(void) {
+	return this->remote_group;
+}
+
+Aca_Security_Group::Aca_Security_Group() {
+}
+
+Aca_Security_Group::Aca_Security_Group(Aca_Security_Group &sg) {
+	this->id = sg.get_id();
+	this->name = sg.get_name();
+	this->format_version = sg.get_format_version();
```

Comment:
format_version will be removed from the schema soon because it is not effective to help with grpc message compatibility (we are keeping the one in the top level GoalState message.